osquery/tables/system/freebsd/process_open_files.cpp

```diff
@@ -0,0 +1,71 @@
+/*
+ *  Copyright (c) 2014, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <stdlib.h>
+#include <unistd.h>
+
+#include <sys/user.h>
+#include <sys/queue.h>
+#include <libprocstat.h>
+
+#include <osquery/core.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+#include <osquery/filesystem.h>
+
+#include "osquery/tables/system/freebsd/procstat.h"
+
+namespace osquery {
+namespace tables {
+
+void genDescriptors(struct procstat* pstat,
```

Comment:
I'm not sure host the procstat APIs work, but in most cases can the `struct procstat*` params be solely input and passed as `const` in these gen functions?